subprojects/instant-execution/src/integTest/groovy/org/gradle/instantexecution/InstantExecutionIncludedBuildChangesIntegrationTest.groovy

```diff
@@ -16,16 +16,55 @@
 
 package org.gradle.instantexecution
 
-
 import org.gradle.api.provider.ValueSource
 import org.gradle.api.provider.ValueSourceParameters
 import org.gradle.instantexecution.fixtures.BuildLogicChangeFixture
+import org.gradle.instantexecution.fixtures.ScriptChangeFixture
+import org.junit.Test
 import spock.lang.Unroll
 
 import static org.junit.Assume.assumeFalse
 
 class InstantExecutionIncludedBuildChangesIntegrationTest extends AbstractInstantExecutionIntegrationTest {
 
+    @Unroll
+    @Test
+    def "invalidates cache upon change to #scriptChangeSpec of included build"() {
+        given:
+        def instant = newInstantExecutionFixture()
+        def fixture = scriptChangeSpec.fixtureForProjectDir(file('build-logic'))
+        fixture.setup()
+        def build = { instantRunLenient(*fixture.buildArguments) }
+        settingsFile << """
+            includeBuild 'build-logic'
+        """
+
+        when:
+        build()
+
+        then:
+        outputContains fixture.expectedOutputBeforeChange
```

Comment:
It would be good to verify that the cache entry is reused prior to making the change.

Same for the other invalidation tests.

Reply:
Done in https://github.com/gradle/gradle/pull/14348.